plugins/image-prioritizer/hooks.php

```diff
@@ -11,3 +11,55 @@
 }
 
 add_action( 'od_init', 'image_prioritizer_init' );
+
+/**
+ * Gets the script to lazy-load videos.
+ *
+ * Load a video and its poster image when it approaches the viewport using an IntersectionObserver.
+ *
+ * Handles 'autoplay' and 'preload' attributes accordingly.
+ *
+ * @since n.e.x.t
+ */
+function image_prioritizer_get_lazy_load_script(): string {
+	return <<<JS
```

Comment:
Eventually should this script and the script in `embed_optimizer_get_lazy_load_script()` be moved to external files that get inlined? This would allow them to be checked with TypeScript as well as allow them to be minified.

Reply:
Yeah a follow-up PR for that would make sense

Reply:
Filed an issue to track this: https://github.com/WordPress/performance/issues/1600

Reply:
Since a contributor already landed this improvement for Embed Optimizer, I've amended this PR with the same change for Image Prioritizer in bde289fc947c99e219a3331edc1f37e66ca974e2